benchmark/CMakeLists.txt

```diff
@@ -4,6 +4,11 @@ if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
         "will be affected")
 endif()
 
+if (GINKGO_BUILD_CUDA AND GINKGO_BUILD_HIP AND GINKGO_HIP_PLATFORM MATCHES "hcc")
```

Comment:
By default benchmarking is on, so in case the user just actually wants to build the library, then there is the option of not building the benchmarks at all. So I think you can mention that `GINKGO_BUILD_BENCHMARKS=off` will also resolve this problem ?

Reply:
Sounds sensible! Just to think ahead: Do you think in the future we will have other parts of Ginkgo that import both HIP and CUDA runtime headers and could cause these issues?

Reply:
If we dont have it now, I dont think we will have it in the future as most of parts of Ginkgo that import the CUDA and HIP runtime stuff is already complete. But I cant be too sure. @yhmtsai is probably in a better position to answer the question.

Reply:
I don't think we will have. Except if it's a new part we develop.

Reply:
I agree with @pratikvn and @tcojean.